### ISSUE TITLE
apply filter_matched on non dropped events

### ISSUE THREAD

**Issue by jsvd** [logstash-plugins/logstash-filter-drop#4]
by default all logstash filters should call `filter_matched(event)` on an event that "passes" the filter.

In this case, an event that is not dropped should be matched, so the "add_tag"/"remove_field"/etc options work.

**Comment by purbon** [comment]
@jsvd can you elaborate more this issue? a description of why you need/think this is important would be nice.

thanks a lot for your time.

**Comment by wiibaa** [comment]
@purbon in fact, with the recent addition of the **percentage** configuration, some events can survive this filter, so it would be meaningful to tag them, no ?

**Comment by guyboertje** [comment]
I agree.  If percentage is less than 100 then any events that are not cancelled should have filter_matched(event) applied to them - you know, like a regular filter would do.

PR with tests most welcome.
